pandas/core/nanops.py

```diff
@@ -829,9 +830,42 @@ def _checked_add_with_arr(arr, b):
 
     Raises
     ------
-    OverflowError if any x + y exceeds the maximum int64 value.
+    OverflowError if any x + y exceeds the maximum or minimum int64 value.
     """
-    if (np.iinfo(np.int64).max - b < arr).any():
-        raise OverflowError("Python int too large to "
-                            "convert to C long")
+    # For performance reasons, we broadcast 'b' to the new array 'b2'
+    # so that it has the same size as 'arr'.
+    if _np_version_under1p10:
+        if lib.isscalar(b):
+            b2 = np.empty(arr.shape)
+            b2.fill(b)
+        else:
+            b2 = b.copy()
```

Comment:
Is the alias a problem? It only exists for the scope of this function? (`broadcast_to` also returns a view, no copy)

Now it just takes an unnecessary copy of an array just to sum it, which has further performance implications (a quick test with the above example codes, it gives me almost 20% of time of the full method)